src/main/kotlin/io/sip3/salto/ce/rtpr/RtprHandler.kt

```diff
@@ -185,55 +207,75 @@ open class RtprHandler : AbstractVerticle() {
         }
     }
 
-    private fun updateWithSdp(packet: Packet, report: RtpReportPayload) {
-        (sdp[sdpSessionId(packet.srcAddr)] ?: sdp[sdpSessionId(packet.dstAddr)])?.let { sdpSession ->
-            report.callId = sdpSession.callId
+    private fun createRtprSession(packet: Packet): RtprSession {
+        val session = RtprSession(packet, rFactorThreshold)
+        (sdp[sdpSessionId(packet.srcAddr)] ?: sdp[sdpSessionId(packet.dstAddr)])?.let { session.sdp = it }
+        return session
+    }
 
-            if (report.source == RtpReportPayload.SOURCE_RTCP && report.duration == 0) {
-                report.duration = report.expectedPacketCount * sdpSession.ptime
-            }
+    private fun sdpSessionId(address: Address): Long {
+        return sdpSessionId(address.addr, address.port)
+    }
+
+    private fun updateWithSdp(report: RtpReportPayload, sdpSession: SdpSession) {
+        report.callId = sdpSession.callId
+
+        if (report.source == RtpReportPayload.SOURCE_RTCP && report.duration == 0) {
+            report.duration = report.expectedPacketCount * sdpSession.ptime
+        }
 
-            sdpSession.codec(report.payloadType.toInt())?.let { codec ->
-                report.codecName = codec.name
+        val codec = if (report.source == RtpReportPayload.SOURCE_RTCP) {
+            sdpSession.codecs.firstOrNull()
+        } else {
+            sdpSession.codec(report.payloadType.toInt())
+        }
 
-                // Raw rFactor value
-                val ppl = report.fractionLost * 100
-                val ieEff = codec.ie + (95 - codec.ie) * ppl / (ppl + codec.bpl)
+        if (codec != null) {
+            report.codecName = codec.name
 
-                report.rFactor = (R0 - ieEff)
+            // Raw rFactor value
+            val ppl = report.fractionLost * 100
+            val ieEff = codec.ie + (95 - codec.ie) * ppl / (ppl + codec.bpl)
 
-                // MoS
-                report.mos = computeMos(report.rFactor)
-            }
+            report.rFactor = (R0 - ieEff)
+
+            // MoS
+            report.mos = computeMos(report.rFactor)
         }
     }
 
-    private fun sdpSessionId(address: Address): Long {
-        return sdpSessionId(address.addr, address.port)
+    private fun sendKeepAlive(session: RtprSession) {
```

Comment:
I think that we can get rid of this method and use `media` endpoint for both - keep-alive and terminated session. We can use the same `RtprSession` object with terminatedAt assigned within terminateExpiredSession...